src/Synthesizer.ml

```diff
@@ -172,13 +179,20 @@ let solve_impl (config : Config.t) (task : task) (stats : stats) =
   let char_candidates = empty_candidates () in
   let string_candidates = empty_candidates () in
   let list_candidates = empty_candidates () in
+  let array_candidates = empty_candidates () in
 
-  let typed_candidates = function
+  let typed_candidates ?(no_tvar = false) = function
     | Type.INT -> int_candidates
     | Type.BOOL -> bool_candidates
     | Type.CHAR -> char_candidates
     | Type.STRING -> string_candidates
     | Type.LIST -> list_candidates
+    | Type.ARRAY (_,_) -> array_candidates
+    | Type.TVAR _ when no_tvar = false -> raise (Internal_Exn "No candidates for TVAR")
+    | Type.TVAR _ -> let (@) = Array.append
+                      in
+                      int_candidates @ array_candidates
+                      (* int_candidates @ bool_candidates @ char_candidates @ string_candidates @ list_candidates @ array_candidates *)
```

Comment:
Uncomment the last line, remove the line above it and shift the last line to the same line as `in`

Also, we should rerun the tests to make sure that we are still able to solve the benchmarks.

Reply:
Changed to append all candidates and ran the following test to verify it works on the benchmarks:  ./scripts/test_all.sh -b benchmarks/LIA/2013.OOPSLA_Hola/. Let me know if you want me to test with other benchmarks.

Reply:
Oh I meant test on your array benchmarks. I would expect the integer benchmarks to still pass, but the array benchmarks may slow down ... hopefully not by a lot